# <div style="text-align: center; background-color:pink; font-family:Georgia, serif; color: black; padding: 20px;line-height: 1;border-radius:5px; border: 2px solid black;">Video Game Sales Analysis: Global Trends and Regional Insights</div>


# Import Library

In [1]:
import pandas as pd
import numpy as np

# About Dataset
* This dataset appears to represent video game sales data, containing information about 16,598 video games. 
* Each row corresponds to a specific video game, and the dataset includes the following columns:

    * Rank: A unique integer identifier that ranks the video games based on a specific criterion (likely sales).
    * Name: The name of the video game.
    * Platform: The gaming platform on which the video game was released (e.g., PS4, Xbox One, PC).
    * Year: The year the game was released. This column has some missing values, with 16,327 non-null entries out of 16,598.
    * Genre: The genre of the video game (e.g., Action, Sports, RPG).
    * Publisher: The company that published the video game. There are a few missing values, with 16,540 non-null entries.
    * NA_Sales: Sales in North America, measured in millions of units.
    * EU_Sales: Sales in Europe, measured in millions of units.
    * JP_Sales: Sales in Japan, measured in millions of units.
    * Other_Sales: Sales in regions other than North America, Europe, and Japan, measured in millions of units.
    * Global_Sales: The total global sales of the video game, measured in millions of units.


# Import data

In [2]:
data=pd.read_csv('Online_game.csv')

In [3]:
data

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

In [4]:
data.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [5]:
data.tail()

Rank                                              Name Platform  \
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year     Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
16593  2002.0  Platform       Kemco      0.01      0.00       0.0   
16594  2003.0   Shooter  Infogrames      0.01      0.00       0.0   
16595  2008.0    Racing  Activision      0.00      0.00       0.0   
16596  2010.0    Puzzle    7G//AMES      0.00      0.01       0.0   
16597  2003.0  Platform     Wanadoo      0.01      0.00       0.0   

       Other_Sales  Global_Sales  
16593          0.0          0.01  
16594          0.0          0.01  
16595          0.0          0.01  
16596          0.0          0.01  
16597          0.0          0.01

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


# Key Observations:
* The dataset is mostly complete, but there are missing values in the Year and Publisher columns.
* Sales data is provided for multiple regions, allowing for regional analysis and comparison of video game popularity.
* The dataset includes both categorical data (e.g., Name, Platform, Genre, Publisher) and numerical data (e.g., Sales figures).

In [7]:
data.describe()

Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16598.000000  16327.000000  16598.000000  16598.000000  16598.000000   
mean    8300.605254   2006.406443      0.264667      0.146652      0.077782   
std     4791.853933      5.828981      0.816683      0.505351      0.309291   
min        1.000000   1980.000000      0.000000      0.000000      0.000000   
25%     4151.250000   2003.000000      0.000000      0.000000      0.000000   
50%     8300.500000   2007.000000      0.080000      0.020000      0.000000   
75%    12449.750000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2020.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16598.000000  16598.000000  
mean       0.048063      0.537441  
std        0.188588      1.555028  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.470000  
max       10.570000     82.740000

In [8]:
data.describe(include='all')

Rank                         Name Platform          Year  \
count   16598.000000                        16598    16598  16327.000000   
unique           NaN                        11493       31           NaN   
top              NaN  Need for Speed: Most Wanted       DS           NaN   
freq             NaN                           12     2163           NaN   
mean     8300.605254                          NaN      NaN   2006.406443   
std      4791.853933                          NaN      NaN      5.828981   
min         1.000000                          NaN      NaN   1980.000000   
25%      4151.250000                          NaN      NaN   2003.000000   
50%      8300.500000                          NaN      NaN   2007.000000   
75%     12449.750000                          NaN      NaN   2010.000000   
max     16600.000000                          NaN      NaN   2020.000000   

         Genre        Publisher      NA_Sales      EU_Sales      JP_Sales  \
count    16598            16540  16598.000000  16598.000000  16598.000000   
unique      12              578           NaN           NaN           NaN   
top     Action  Electronic Arts           NaN           NaN           NaN   
freq      3316             1351           NaN           NaN           NaN   
mean       NaN              NaN      0.264667      0.146652      0.077782   
std        NaN              NaN      0.816683      0.505351      0.309291   
min        NaN              NaN      0.000000      0.000000      0.000000   
25%        NaN              NaN      0.000000      0.000000      0.000000   
50%        NaN              NaN      0.080000      0.020000      0.000000   
75%        NaN              NaN      0.240000      0.110000      0.040000   
max        NaN              NaN     41.490000     29.020000     10.220000   

         Other_Sales  Global_Sales  
count   16598.000000  16598.000000  
unique           NaN           NaN  
top              NaN           NaN  
freq             NaN           NaN  
mean        0.048063      0.537441  
std         0.188588      1.555028  
min         0.000000      0.010000  
25%         0.000000      0.060000  
50%         0.010000      0.170000  
75%         0.040000      0.470000  
max        10.570000     82.740000

In [9]:
data.describe(include='object')

Name Platform   Genre        Publisher
count                         16598    16598   16598            16540
unique                        11493       31      12              578
top     Need for Speed: Most Wanted       DS  Action  Electronic Arts
freq                             12     2163    3316             1351

# Checking Null values

In [10]:
data.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

* Data have null values at feature year

# Identifing number of unique values

In [11]:
data.nunique()

Rank            16598
Name            11493
Platform           31
Year               39
Genre              12
Publisher         578
NA_Sales          409
EU_Sales          305
JP_Sales          244
Other_Sales       157
Global_Sales      623
dtype: int64

# Checking duplicate values in data

In [12]:
data.duplicated().sum()

0

In [13]:
data[data.duplicated()]

Empty DataFrame
Columns: [Rank, Name, Platform, Year, Genre, Publisher, NA_Sales, EU_Sales, JP_Sales, Other_Sales, Global_Sales]
Index: []

* No duplicate values

In [14]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


# How many unique video game titles are there in the dataset?

In [15]:
data['Name'].unique()

array(['Wii Sports', 'Super Mario Bros.', 'Mario Kart Wii', ...,
       'Plushees', 'Woody Woodpecker in Crazy Castle 5', 'Know How 2'],
      dtype=object)

In [16]:
data['Name'].nunique()

11493

* In this dataset contain 11493 unique vedio games 

# Which gaming platform has the most titles?

In [17]:
data.groupby('Platform')['Name'].count().sort_values(ascending=False).head()

Platform
DS      2163
PS2     2161
PS3     1329
Wii     1325
X360    1265
Name: Name, dtype: int64

In [18]:
data['Platform'].value_counts().idxmax()

'DS'

* DS platform has the most title

# What is the most common genre of video games?

In [19]:
data['Genre'].value_counts().sort_values(ascending=False)

Action          3316
Sports          2346
Misc            1739
Role-Playing    1488
Shooter         1310
Adventure       1286
Racing          1249
Platform         886
Simulation       867
Fighting         848
Strategy         681
Puzzle           582
Name: Genre, dtype: int64

In [20]:
data['Genre'].value_counts().sort_values(ascending=False).idxmax()

'Action'

* Action type of games are more common genre

# How many missing values are there in the Year and Publisher columns?

In [21]:
print(f'Missing values in Year :',data['Year'].isnull().sum())
print(f'Missing values in Publisher :',data['Publisher'].isnull().sum())

Missing values in Year : 271
Missing values in Publisher : 58


# How can you fill the missing values in the Year column?

In [22]:
data['Year'].fillna(data['Year'].median(),inplace=True)

# How can you deal with the missing Publisher values?

In [23]:
data['Publisher'].fillna('Unknown',inplace=True)

# Convert year to integer

In [24]:
data['Year']=data['Year'].astype('int64')

In [25]:
data.isnull().sum()

Rank            0
Name            0
Platform        0
Year            0
Genre           0
Publisher       0
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64

# Sales Analysis

# Which video game has the highest global sales?

In [26]:
data.groupby('Name')['Global_Sales'].max().sort_values(ascending=False)

Name
Wii Sports                                        82.74
Super Mario Bros.                                 40.24
Mario Kart Wii                                    35.82
Wii Sports Resort                                 33.00
Pokemon Red/Pokemon Blue                          31.37
                                                  ...  
Muv-Luv Alternative                                0.01
Bullet Soul: Tama Tamashii                         0.01
Miyako: Awayuki no Utage                           0.01
Space Hulk                                         0.01
Horse Life 4: My Horse, My Friend, My Champion     0.01
Name: Global_Sales, Length: 11493, dtype: float64

In [27]:
data.groupby('Name')['Global_Sales'].max().sort_values(ascending=False).idxmax()

'Wii Sports'

In [28]:
data.loc[data['Global_Sales'].idxmax()]

Rank                     1
Name            Wii Sports
Platform               Wii
Year                  2006
Genre               Sports
Publisher         Nintendo
NA_Sales             41.49
EU_Sales             29.02
JP_Sales              3.77
Other_Sales           8.46
Global_Sales         82.74
Name: 0, dtype: object

* Wii Sports has the highest global sales

# How do sales compare across different regions (NA, EU, JP, Other)?

In [29]:
data[['NA_Sales','EU_Sales','JP_Sales','Other_Sales']].mean()

NA_Sales       0.264667
EU_Sales       0.146652
JP_Sales       0.077782
Other_Sales    0.048063
dtype: float64

In [30]:
data.groupby('Year')[['NA_Sales','EU_Sales','JP_Sales','Other_Sales']].mean()

NA_Sales  EU_Sales  JP_Sales  Other_Sales
Year                                           
1980  1.176667  0.074444  0.000000     0.013333
1981  0.726087  0.042609  0.000000     0.006957
1982  0.747778  0.045833  0.000000     0.008611
1983  0.456471  0.047059  0.476471     0.008235
1984  2.377143  0.150000  1.019286     0.050000
1985  2.409286  0.338571  1.040000     0.065714
1986  0.595238  0.135238  0.943333     0.091905
1987  0.528750  0.088125  0.726875     0.012500
1988  1.591333  0.439333  1.050667     0.066000
1989  2.655882  0.496471  1.080000     0.088235
1990  1.591250  0.476875  0.930000     0.087500
1991  0.311220  0.096341  0.360488     0.018049
1992  0.787674  0.272326  0.672326     0.038372
1993  0.252000  0.077500  0.422167     0.014833
1994  0.232645  0.122975  0.280909     0.018182
1995  0.113333  0.068037  0.208904     0.012055
1996  0.329886  0.179696  0.218403     0.029240
1997  0.327855  0.167197  0.169100     0.031592
1998  0.338681  0.176517  0.132032     0.029103
1999  0.372959  0.185414  0.154852     0.029734
2000  0.270745  0.151146  0.122550     0.033295
2001  0.360954  0.196867  0.082697     0.047220
2002  0.260784  0.132376  0.050374     0.032907
2003  0.249794  0.133948  0.044129     0.033561
2004  0.291730  0.140655  0.054587     0.061979
2005  0.257821  0.129586  0.057683     0.043124
2006  0.261032  0.128214  0.073145     0.053998
2007  0.252254  0.125940  0.045492     0.058615
2008  0.246106  0.129132  0.042199     0.057696
2009  0.236792  0.133885  0.043249     0.052250
2010  0.241652  0.140373  0.047252     0.047577
2011  0.211642  0.147006  0.046567     0.047752
2012  0.235860  0.180791  0.078752     0.057565
2013  0.283462  0.230403  0.087161     0.072930
2014  0.226753  0.215893  0.067801     0.068763
2015  0.167459  0.159137  0.054919     0.048876
2016  0.065872  0.077791  0.039826     0.022529
2017  0.000000  0.000000  0.016667     0.000000
2020  0.270000  0.000000  0.000000     0.020000

# What is the average sales per genre?

In [54]:
data.groupby(['Genre'])['Global_Sales'].mean().sort_values(ascending=False)

Genre
Platform        0.938341
Shooter         0.791885
Role-Playing    0.623233
Racing          0.586101
Sports          0.567319
Fighting        0.529375
Action          0.528100
Misc            0.465762
Simulation      0.452364
Puzzle          0.420876
Strategy        0.257151
Adventure       0.185879
Name: Global_Sales, dtype: float64

# How have global sales trended over the years?

In [56]:
data.groupby('Year')['Global_Sales'].mean()

Year
1980    1.264444
1981    0.777609
1982    0.801667
1983    0.987647
1984    3.597143
1985    3.852857
1986    1.765238
1987    1.358750
1988    3.148000
1989    4.320588
1990    3.086875
1991    0.786098
1992    1.771163
1993    0.766333
1994    0.654298
1995    0.402329
1996    0.757224
1997    0.695433
1998    0.676702
1999    0.743402
2000    0.577536
2001    0.687697
2002    0.477105
2003    0.461742
2004    0.549554
2005    0.488778
2006    0.516905
2007    0.482831
2008    0.475420
2009    0.466317
2010    0.476926
2011    0.453020
2012    0.553333
2013    0.674194
2014    0.579124
2015    0.430684
2016    0.206192
2017    0.016667
2020    0.290000
Name: Global_Sales, dtype: float64

In [33]:
data.groupby('Year')['Global_Sales'].sum()

Year
1980     11.38
1981     35.77
1982     28.86
1983     16.79
1984     50.36
1985     53.94
1986     37.07
1987     21.74
1988     47.22
1989     73.45
1990     49.39
1991     32.23
1992     76.16
1993     45.98
1994     79.17
1995     88.11
1996    199.15
1997    200.98
1998    256.47
1999    251.27
2000    201.56
2001    331.47
2002    395.52
2003    357.85
2004    419.31
2005    459.94
2006    521.04
2007    711.21
2008    678.90
2009    667.30
2010    600.45
2011    515.99
2012    363.54
2013    368.11
2014    337.05
2015    264.44
2016     70.93
2017      0.05
2020      0.29
Name: Global_Sales, dtype: float64

# Video games released after 2010

In [34]:
data_2010=data[data['Year']>2010].reset_index(drop=True)
data_2010

Rank                                               Name Platform  Year  \
0        17                                 Grand Theft Auto V      PS3  2013   
1        24                                 Grand Theft Auto V     X360  2013   
2        30                     Call of Duty: Modern Warfare 3     X360  2011   
3        33                                Pokemon X/Pokemon Y      3DS  2013   
4        34                          Call of Duty: Black Ops 3      PS4  2015   
...     ...                                                ...      ...   ...   
3881  16579                                  Rugby Challenge 3     XOne  2016   
3882  16581                      Outdoors Unleashed: Africa 3D      3DS  2011   
3883  16584                                          Fit & Fun      Wii  2011   
3884  16588                                             Breach       PC  2011   
3885  16592  Chou Ezaru wa Akai Hana: Koi wa Tsuki ni Shiru...      PSV  2016   

             Genre             Publisher  NA_Sales  EU_Sales  JP_Sales  \
0           Action  Take-Two Interactive      7.01      9.27      0.97   
1           Action  Take-Two Interactive      9.63      5.31      0.06   
2          Shooter            Activision      9.03      4.28      0.13   
3     Role-Playing              Nintendo      5.17      4.05      4.34   
4          Shooter            Activision      5.77      5.81      0.35   
...            ...                   ...       ...       ...       ...   
3881        Sports  Alternative Software      0.00      0.01      0.00   
3882        Sports               Mastiff      0.01      0.00      0.00   
3883        Sports               Unknown      0.00      0.01      0.00   
3884       Shooter             Destineer      0.01      0.00      0.00   
3885        Action       dramatic create      0.00      0.00      0.01   

      Other_Sales  Global_Sales  
0            4.14         21.40  
1            1.38         16.38  
2            1.32         14.76  
3            0.79         14.35  
4            2.31         14.24  
...           ...           ...  
3881         0.00          0.01  
3882         0.00          0.01  
3883         0.00          0.01  
3884         0.00          0.01  
3885         0.00          0.01  

[3886 rows x 11 columns]

# Global_Sales in descending order

In [35]:
sort_gb=data.sort_values(by='Global_Sales',ascending=False).reset_index(drop=True)
sort_gb

Rank                                  Name Platform  Year  \
0          1                            Wii Sports      Wii  2006   
1          2                     Super Mario Bros.      NES  1985   
2          3                        Mario Kart Wii      Wii  2008   
3          4                     Wii Sports Resort      Wii  2009   
4          5              Pokemon Red/Pokemon Blue       GB  1996   
...      ...                                   ...      ...   ...   
16593  16189                           BattleForge       PC  2009   
16594  16190                        Jewel Quest II       PC  2007   
16595  16191                      Toro to Morimori      PS3  2009   
16596  16192  Sonic & All-Stars Racing Transformed       PC  2013   
16597  16600                      Spirits & Spells      GBA  2003   

              Genre                    Publisher  NA_Sales  EU_Sales  \
0            Sports                     Nintendo     41.49     29.02   
1          Platform                     Nintendo     29.08      3.58   
2            Racing                     Nintendo     15.85     12.88   
3            Sports                     Nintendo     15.75     11.01   
4      Role-Playing                     Nintendo     11.27      8.89   
...             ...                          ...       ...       ...   
16593      Strategy              Electronic Arts      0.00      0.01   
16594        Puzzle                    Avanquest      0.00      0.01   
16595          Misc  Sony Computer Entertainment      0.00      0.00   
16596        Racing                         Sega      0.00      0.01   
16597      Platform                      Wanadoo      0.01      0.00   

       JP_Sales  Other_Sales  Global_Sales  
0          3.77         8.46         82.74  
1          6.81         0.77         40.24  
2          3.79         3.31         35.82  
3          3.28         2.96         33.00  
4         10.22         1.00         31.37  
...         ...          ...           ...  
16593      0.00         0.00          0.01  
16594      0.00         0.00          0.01  
16595      0.01         0.00          0.01  
16596      0.00         0.00          0.01  
16597      0.00         0.00          0.01  

[16598 rows x 11 columns]

# Find the top 10 games in the Action genre by global sales?

In [36]:
action_data=data[data['Genre'].isin(['Action'])]
action_data.sort_values(by='Global_Sales',ascending=False).head(10).reset_index(drop=True)

Rank                                  Name Platform  Year   Genre  \
0    17                    Grand Theft Auto V      PS3  2013  Action   
1    18         Grand Theft Auto: San Andreas      PS2  2004  Action   
2    24                    Grand Theft Auto V     X360  2013  Action   
3    25           Grand Theft Auto: Vice City      PS2  2002  Action   
4    39                  Grand Theft Auto III      PS2  2001  Action   
5    45                    Grand Theft Auto V      PS4  2014  Action   
6    46  Pokemon HeartGold/Pokemon SoulSilver       DS  2009  Action   
7    52                   Grand Theft Auto IV     X360  2008  Action   
8    57                   Grand Theft Auto IV      PS3  2008  Action   
9    83                        FIFA Soccer 13      PS3  2012  Action   

              Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
0  Take-Two Interactive      7.01      9.27      0.97         4.14   
1  Take-Two Interactive      9.43      0.40      0.41        10.57   
2  Take-Two Interactive      9.63      5.31      0.06         1.38   
3  Take-Two Interactive      8.41      5.49      0.47         1.78   
4  Take-Two Interactive      6.99      4.51      0.30         1.30   
5  Take-Two Interactive      3.80      5.81      0.36         2.02   
6              Nintendo      4.40      2.77      3.96         0.77   
7  Take-Two Interactive      6.76      3.10      0.14         1.03   
8  Take-Two Interactive      4.76      3.76      0.44         1.62   
9       Electronic Arts      1.06      5.05      0.13         2.01   

   Global_Sales  
0         21.40  
1         20.81  
2         16.38  
3         16.15  
4         13.10  
5         11.98  
6         11.90  
7         11.02  
8         10.57  
9          8.24

# Find the top 10 games in the Adventure genre by global sales?

In [37]:
Adventure_data=data[data['Genre'].isin(['Adventure'])]
Adventure_data.sort_values(by='Global_Sales',ascending=False).head(10).reset_index(drop=True)

Rank                                Name Platform  Year      Genre  \
0    51  Super Mario Land 2: 6 Golden Coins       GB  1992  Adventure   
1   159                    Assassin's Creed     X360  2007  Adventure   
2   219                    Assassin's Creed      PS3  2007  Adventure   
3   252     Zelda II: The Adventure of Link      NES  1987  Adventure   
4   401          Rugrats: Search For Reptar       PS  1998  Adventure   
5   418                          L.A. Noire      PS3  2011  Adventure   
6   435   Club Penguin: Elite Penguin Force       DS  2008  Adventure   
7   448                          Heavy Rain      PS3  2010  Adventure   
8   522                                Myst       PC  1994  Adventure   
9   550                          L.A. Noire     X360  2011  Adventure   

                     Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
0                     Nintendo      6.16      2.04      2.69         0.29   
1                      Ubisoft      3.28      1.65      0.07         0.55   
2                      Ubisoft      1.91      2.00      0.09         0.83   
3                     Nintendo      2.19      0.50      1.61         0.08   
4                          THQ      1.63      1.53      0.00         0.18   
5         Take-Two Interactive      1.27      1.33      0.12         0.51   
6   Disney Interactive Studios      1.88      0.98      0.00         0.30   
7  Sony Computer Entertainment      1.29      1.27      0.06         0.50   
8                      Red Orb      0.02      2.79      0.00         0.00   
9         Take-Two Interactive      1.52      0.94      0.02         0.24   

   Global_Sales  
0         11.18  
1          5.55  
2          4.83  
3          4.38  
4          3.34  
5          3.23  
6          3.16  
7          3.12  
8          2.81  
9          2.72

In [38]:
data['Genre'].unique()

array(['Sports', 'Platform', 'Racing', 'Role-Playing', 'Puzzle', 'Misc',
       'Shooter', 'Simulation', 'Action', 'Fighting', 'Adventure',
       'Strategy'], dtype=object)

# What are the total global sales per genre?

In [57]:
data.groupby('Genre')['Global_Sales'].sum().sort_values(ascending=False)

Genre
Action          1751.18
Sports          1330.93
Shooter         1037.37
Role-Playing     927.37
Platform         831.37
Misc             809.96
Racing           732.04
Fighting         448.91
Simulation       392.20
Puzzle           244.95
Adventure        239.04
Strategy         175.12
Name: Global_Sales, dtype: float64

# Which publisher has the highest total sales in North America?

In [40]:
data.groupby('Publisher')['NA_Sales'].sum().sort_values(ascending=False)

Publisher
Nintendo                       816.87
Electronic Arts                595.07
Activision                     429.70
Sony Computer Entertainment    265.22
Ubisoft                        253.43
                                ...  
Graphsim Entertainment           0.00
Revolution (Japan)               0.00
Revolution Software              0.00
Grand Prix Games                 0.00
Seventh Chord                    0.00
Name: NA_Sales, Length: 578, dtype: float64

In [41]:
data.groupby('Publisher')['NA_Sales'].sum().idxmax()

'Nintendo'

# Which publisher has the highest total sales in Europe?

In [42]:
data.groupby('Publisher')['EU_Sales'].sum().idxmax()

'Nintendo'

# Which publisher has the maximum sales in Europe?

In [60]:
data.groupby('Publisher')['EU_Sales'].max().idxmax()

'Nintendo'

# Which publisher has the highest total sales in Japan?

In [43]:
data.groupby('Publisher')['JP_Sales'].sum().idxmax()

'Nintendo'

# Which publisher has the highest total sales in Global_Sales?

In [61]:
data.groupby('Publisher')['Global_Sales'].sum().idxmax()

'Nintendo'

# Which publisher has the highest total sales in other than North America, Europe, and Japan?

In [44]:
data.groupby('Publisher')['Other_Sales'].sum().idxmax()

'Electronic Arts'

### Best Publishers in Different Regions:

* Europe, Japan, and North America:
  * Nintendo is the top-performing publisher across these regions. It has the highest total sales in Europe, Japan, and North America, showcasing its dominance in these key gaming markets.

* Other Regions (excluding North America, Europe, and Japan):
  * Electronic Arts (EA) leads in the regions outside of North America, Europe, and Japan. EA has the highest total sales in these other markets, making it the best publisher in these regions.



# What is the average sales per platform?

In [45]:
data.groupby('Platform')['Global_Sales'].mean()

Platform
2600    0.729925
3DO     0.033333
3DS     0.486169
DC      0.307115
DS      0.380254
GB      2.606633
GBA     0.387470
GC      0.358561
GEN     1.050370
GG      0.040000
N64     0.686144
NES     2.561939
NG      0.120000
PC      0.269604
PCFX    0.030000
PS      0.610920
PS2     0.581046
PS3     0.720722
PS4     0.827679
PSP     0.244254
PSV     0.149952
SAT     0.194162
SCD     0.311667
SNES    0.837029
TG16    0.080000
WS      0.236667
Wii     0.699404
WiiU    0.572448
X360    0.774672
XB      0.313422
XOne    0.662254
Name: Global_Sales, dtype: float64

#  Total sales by genre and platform?

In [46]:
data.pivot_table(index='Genre',columns='Platform',values='Global_Sales',aggfunc='sum')

Platform       2600   3DO    3DS    DC      DS     GB    GBA     GC    GEN  \
Genre                                                                        
Action        29.34   NaN  57.02  1.26  115.56   7.92  55.76  37.84   2.74   
Adventure      1.70  0.06   4.81  2.50   47.29  17.16  14.68   5.93   0.19   
Fighting       1.24   NaN  10.46  1.83    7.20    NaN   4.21  18.43   5.90   
Misc           3.58   NaN  10.48   NaN  137.76  13.35  36.25  16.73   0.03   
Platform      13.27   NaN  32.23  2.54   77.45  54.91  78.30  28.66  15.45   
Puzzle        14.68  0.02   5.57   NaN   84.29  47.47  12.92   4.70    NaN   
Racing         2.91   NaN  14.49  2.65   38.64   4.55  18.80  21.89   0.26   
Role-Playing    NaN   NaN  75.74  0.68  126.85  88.24  64.21  13.15   0.27   
Shooter       26.48   NaN   1.29  0.33    8.20   1.20   3.60  13.63   0.13   
Simulation     0.45  0.02  27.08  0.52  132.03   3.55   5.91   8.59    NaN   
Sports         3.43   NaN   6.20  3.66   31.83   9.05  16.41  25.49   3.20   
Strategy        NaN   NaN   2.09   NaN   15.39   8.05   7.45   4.32   0.19   

Platform        GG  ...   SAT   SCD   SNES  TG16    WS     Wii   WiiU    X360  \
Genre               ...                                                         
Action         NaN  ...  0.65   NaN  10.08   NaN   NaN  118.58  19.35  242.67   
Adventure      NaN  ...  4.16   NaN   1.50  0.14   NaN   18.43   0.17   15.23   
Fighting       NaN  ...  8.52   NaN  26.95   NaN   NaN   23.86   6.36   37.64   
Misc           NaN  ...  1.20  0.10   5.02   NaN   NaN  221.06  12.23   91.96   
Platform      0.04  ...  0.76  1.50  65.65   NaN   NaN   90.74  21.24   11.39   
Puzzle         NaN  ...  1.00   NaN   6.38   NaN   NaN   15.67   1.33    0.85   
Racing         NaN  ...  2.40  0.07  13.49   NaN   NaN   61.28   7.77   65.99   
Role-Playing   NaN  ...  3.76  0.06  36.43   NaN  1.22   14.06   2.47   71.98   
Shooter        NaN  ...  3.98   NaN   6.07  0.02   NaN   28.77   6.17  278.55   
Simulation     NaN  ...  1.13   NaN   5.63   NaN   NaN   36.97   0.21   14.45   
Sports         NaN  ...  2.79   NaN  17.87   NaN   NaN  292.06   3.32  139.12   
Strategy       NaN  ...  3.24  0.14   4.98   NaN  0.20    5.23   1.24   10.13   

Platform         XB   XOne  
Genre                       
Action        49.28  33.79  
Adventure      3.06   2.51  
Fighting      13.55   2.31  
Misc           9.58   6.86  
Platform       9.66   0.81  
Puzzle         0.42    NaN  
Racing        31.49   8.80  
Role-Playing  13.51   9.48  
Shooter       63.55  51.61  
Simulation     7.11   0.54  
Sports        54.27  23.97  
Strategy       2.78   0.38  

[12 rows x 31 columns]

In [47]:
data.pivot_table(columns='Genre',index='Platform',values='Global_Sales',aggfunc='sum')

Genre     Action  Adventure  Fighting    Misc  Platform  Puzzle  Racing  \
Platform                                                                  
2600       29.34       1.70      1.24    3.58     13.27   14.68    2.91   
3DO          NaN       0.06       NaN     NaN       NaN    0.02     NaN   
3DS        57.02       4.81     10.46   10.48     32.23    5.57   14.49   
DC          1.26       2.50      1.83     NaN      2.54     NaN    2.65   
DS        115.56      47.29      7.20  137.76     77.45   84.29   38.64   
GB          7.92      17.16       NaN   13.35     54.91   47.47    4.55   
GBA        55.76      14.68      4.21   36.25     78.30   12.92   18.80   
GC         37.84       5.93     18.43   16.73     28.66    4.70   21.89   
GEN         2.74       0.19      5.90    0.03     15.45     NaN    0.26   
GG           NaN        NaN       NaN     NaN      0.04     NaN     NaN   
N64        29.58       0.45     22.08   11.19     36.13    3.41   40.09   
NES        28.75       4.38      6.54    3.59     95.78   21.00    9.78   
NG           NaN        NaN      1.42     NaN       NaN     NaN     NaN   
PC         31.53      10.09      0.27    8.56      0.49    0.92    4.06   
PCFX         NaN        NaN       NaN     NaN       NaN     NaN     NaN   
PS        127.05      20.97     72.68   44.90     64.21   12.08  103.19   
PS2       272.76      21.16     92.60  101.14     72.51    5.90  156.28   
PS3       307.88      22.90     51.70   47.85     30.33    0.60   73.95   
PS4        87.06       4.70      8.04    7.40      7.01    0.02   11.53   
PSP        64.72      10.69     21.97   13.96     17.28    5.52   34.73   
PSV        20.01       4.18      3.15    5.18      3.03    0.20    2.95   
SAT         0.65       4.16      8.52    1.20      0.76    1.00    2.40   
SCD          NaN        NaN       NaN    0.10      1.50     NaN    0.07   
SNES       10.08       1.50     26.95    5.02     65.65    6.38   13.49   
TG16         NaN       0.14       NaN     NaN       NaN     NaN     NaN   
WS           NaN        NaN       NaN     NaN       NaN     NaN     NaN   
Wii       118.58      18.43     23.86  221.06     90.74   15.67   61.28   
WiiU       19.35       0.17      6.36   12.23     21.24    1.33    7.77   
X360      242.67      15.23     37.64   91.96     11.39    0.85   65.99   
XB         49.28       3.06     13.55    9.58      9.66    0.42   31.49   
XOne       33.79       2.51      2.31    6.86      0.81     NaN    8.80   

Genre     Role-Playing  Shooter  Simulation  Sports  Strategy  
Platform                                                       
2600               NaN    26.48        0.45    3.43       NaN  
3DO                NaN      NaN        0.02     NaN       NaN  
3DS              75.74     1.29       27.08    6.20      2.09  
DC                0.68     0.33        0.52    3.66       NaN  
DS              126.85     8.20      132.03   31.83     15.39  
GB               88.24     1.20        3.55    9.05      8.05  
GBA              64.21     3.60        5.91   16.41      7.45  
GC               13.15    13.63        8.59   25.49      4.32  
GEN               0.27     0.13         NaN    3.20      0.19  
GG                 NaN      NaN         NaN     NaN       NaN  
N64               3.36    19.27       10.19   32.80     10.33  
NES              18.78    35.62         NaN   26.85       NaN  
NG                 NaN      NaN         NaN    0.02       NaN  
PC               47.78    45.32       51.91   12.01     45.88  
PCFX              0.03      NaN         NaN     NaN       NaN  
PS               78.30    39.31       25.33  120.97     21.67  
PS2              93.50   108.57       42.66  273.41     15.15  
PS3              75.30   196.04       10.79  135.56      4.94  
PS4              25.77    75.32        0.77   50.07      0.41  
PSP              49.05    19.77        6.28   41.69     10.62  
PSV              13.39     4.57        0.08    4.88      0.31  
SAT               3.76     3.98        1.13    2.79 

# What is the median sales value per year for each platform?

In [48]:
data.groupby(['Year','Platform'])['Global_Sales'].median()

Year  Platform
1980  2600        0.770
1981  2600        0.465
1982  2600        0.540
1983  2600        0.460
      NES         1.635
                  ...  
2016  X360        0.065
      XOne        0.065
2017  PS4         0.030
      PSV         0.010
2020  DS          0.290
Name: Global_Sales, Length: 247, dtype: float64

In [49]:
data.pivot_table(index='Platform',columns='Year',values='Global_Sales',aggfunc='median')

Year      1980   1981  1982   1983  1984  1985  1986   1987   1988   1989  \
Platform                                                                    
2600      0.77  0.465  0.54  0.460  0.27  0.45  0.33  0.355  0.375  0.310   
3DO        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
3DS        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
DC         NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
DS         NaN    NaN   NaN    NaN   NaN  0.02   NaN    NaN    NaN    NaN   
GB         NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN  1.430  1.965   
GBA        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
GC         NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
GEN        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
GG         NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
N64        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
NES        NaN    NaN   NaN  1.635  1.52  1.05  1.50  1.785  2.140  1.100   
NG         NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
PC         NaN    NaN   NaN    NaN   NaN  0.03   NaN    NaN  0.030    NaN   
PCFX       NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
PS         NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
PS2        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
PS3        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
PS4        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
PSP        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
PSV        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
SAT        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
SCD        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
SNES       NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
TG16       NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
WS         NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
Wii        NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
WiiU       NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
X360       NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
XB         NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   
XOne       NaN    NaN   NaN    NaN   NaN   NaN   NaN    NaN    NaN    NaN   

Year      ...   2009   2010   2011   2012   2013   2014   2015   2016  2017  \
Platform  ...                                                                 
2600      ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
3DO       ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
3DS       ...    NaN    NaN  0.145  0.190  0.100  0.090  0.085  0.070   NaN   
DC        ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
DS        ...  0.110  0.110  0.090  0.030  0.150  0.020    NaN    NaN   NaN   
GB        ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
GBA       ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
GC        ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
GEN       ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
GG        ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
N64       ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
NES       ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
NG        ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
PC        ...  0.030  0.060  0.080  0.120  0.180  0.135  0.080  0.030   NaN   
PCFX      ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
PS        ...    NaN    NaN    NaN    NaN    NaN    NaN    NaN    NaN   NaN   
PS2       ...  0.170  0.055  0.060  

# Market share by publisher?

In [50]:
data.groupby('Publisher')['Global_Sales'].sum()

Publisher
10TACLE Studios                  0.11
1C Company                       0.10
20th Century Fox Video Games     1.94
2D Boy                           0.04
3DO                             10.12
                                ...  
id Software                      0.03
imageepoch Inc.                  0.04
inXile Entertainment             0.10
mixi, Inc                        0.86
responDESIGN                     0.13
Name: Global_Sales, Length: 578, dtype: float64

# What is the correlation between North American and European sales?

In [51]:
correlation_na_eu = data['NA_Sales'].corr(data['EU_Sales'])
correlation_na_eu

0.7677267483702562

# Categorizes games into high, medium, and low sales based on their global sales figures?

In [52]:
data['Sales_Category'] = pd.cut(data['Global_Sales'], bins=[0, 1, 5, data['Global_Sales'].max()], labels=['Low', 'Medium', 'High'])


In [53]:
data.head()

Rank                      Name Platform  Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales Sales_Category  
0     41.49     29.02      3.77         8.46         82.74           High  
1     29.08      3.58      6.81         0.77         40.24           High  
2     15.85     12.88      3.79         3.31         35.82           High  
3     15.75     11.01      3.28         2.96         33.00           High  
4     11.27      8.89     10.22         1.00         31.37           High